### PR TITLE
Move 3 container checks to crane

### LINKED ISSUE #202: Migrate LayerCountAcceptable Check to Crane
## Bug Description
Migrate LayerCountAcceptable Check to Crane

## Version and Command Invocation
NA

## Steps to Reproduce:
NA

## Expected Result
NA

## Actual Result
NA

## Additional Context
NA

### PR DESCRIPTION
**Pull request by bcrochet**
Moves 3 container checks:
Max layers
Prohibited packages
Required Labels

Fixes #202

### CODE REVIEW

File: certification/internal/policy/container/has_required_labels.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+)
+
+var requiredLabels = []string{"name", "vendor", "version", "release", "summary", "description"}
+
+// HasRequiredLabelsCheck evaluates the image manifest to ensure that the appropriate metadata
+// labels are present on the image asset as it exists in its current container registry.
+type HasRequiredLabelsCheck struct{}
+
+func (p *HasRequiredLabelsCheck) Validate(imgRef certification.ImageReference) (bool, error) {
+	// TODO: if we're going have the image json on disk already, we should use it here instead of podman inspect-ing.
```
Comment (samira-barouti) [inline certification/internal/policy/container/has_required_labels.go:16]: This TODO could be removed now \o/

File: certification/internal/policy/container/less_than_max_layers.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+const (
+	acceptableLayerMax = 40
+)
+
+// UnderLayerMaxCheck ensures that the image has less layers in its assembly than a predefined maximum.
+type UnderLayerMaxCheck struct{}
+
+func (p *UnderLayerMaxCheck) Validate(imageRef certification.ImageReference) (bool, error) {
+	// TODO: if we're going have the image json on disk already, we should use it here instead of podman inspect-ing.
```
Comment (samira-barouti) [inline certification/internal/policy/container/less_than_max_layers.go:19]: Same goes here

**Review by samira-barouti — APPROVED** [review]
Two small nits... otherwise good to go

**Review by samira-barouti — APPROVED** [review]
/lgtm

File: certification/internal/policy/container/less_than_max_layers.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	acceptableLayerMax = 40
+)
+
+// UnderLayerMaxCheck ensures that the image has less layers in its assembly than a predefined maximum.
+type UnderLayerMaxCheck struct{}
```
Comment (acornett21) [inline certification/internal/policy/container/less_than_max_layers.go:16]: I think the file name and the check name are confusing for this check, and this one doesn't seem to follow the conventions of the other one. However of all the check, the rest have more natural names by the nature of what the check is doing.

File: certification/internal/policy/container/has_prohibited_packages.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		}
+	}
+
+	log.Warn("The number of prohibited package found in the container image: ", len(prohibitedPackages))
+	if len(prohibitedPackages) == 0 {
+		return true, nil
+	}
+
+	log.Warn("found the following prohibited packages: ", prohibitedPackages)
+	return false, nil
```
Comment (samira-barouti) [inline certification/internal/policy/container/has_prohibited_packages.go:64]: I'd probably rewrite this as follow to be consistent with the required label check:
```
if len(prohibitedPackages) > 0 {
    log.Warn("The number of prohibited package found in the container image: ", len(prohibitedPackages))
    log.Warn("found the following prohibited packages: ", prohibitedPackages)
}
return len(prohibitedPackages) == 0, nil
```

File: certification/internal/policy/container/has_prohibited_packages.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+		return nil, err
+	}
+	pkgList, err := db.ListPackages()
+	if err != nil {
+		return nil, err
+	}
+	pkgs := make([]string, len(pkgList))
+	for i, pkg := range pkgList {
+		pkgs[i] = pkg.Name
+	}
```
[the remark this answers is not in the document]
Comment (bcrochet) [reply certification/internal/policy/container/has_prohibited_packages.go:38]: So, there are two reasons that I implemented it this way. One, I did not want to leak the data structure returned by db.ListPackages, so iterating over that list to generate a string slice was done. Two, supporting the glob. By iterating over the list of packages (as opposed to iterating over the prohibited packages), I can also check the list of globs and see if the package name starts with anything in the glob list (which happens to be just one atm). There is no operator to see if a key starts with a value (without iterating over the map). So I don't think this optimization would save us anything. If anything, we'd be doing more operations. I see where you're going, and if it weren't for the glob, I'd agree.

[the remark this answers is not in the document]
Comment (samira-barouti) [reply certification/internal/policy/container/has_prohibited_packages.go:38]: Makes sense... I overlooked the partial matching that needs to be done for the glob.

RE the other point, I still see value in merging getDataToValidate() and validate() and doing one iteration; but if the number of container packages is not huge, the extra iteration is probably negligible in terms of elapsed time.

**Review by samira-barouti — APPROVED** [review]
/lgtm

**Pull request merged**